assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.js

```diff
@@ -0,0 +1,71 @@
+import Component from "@ember/component";
+import { isBlank } from "@ember/utils";
+import { action, computed } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class MoveToChannelModalInner extends Component {
+  @service chat;
+  @service router;
+  tagName = "";
+  sourceChannel = null;
+  destinationChannelId = null;
+  selectedMessageIds = null;
+  channels = null;
+
+  didInsertElement() {
+    this._super(...arguments);
+    this._getInitialChannels();
+  }
+
+  _getInitialChannels() {
+    return this.chat.getChannels().then((channels) => {
+      this.set("channels", channels);
+    });
+  }
+
+  @computed("destinationChannelId")
+  get disableMoveButton() {
+    return isBlank(this.destinationChannelId);
+  }
+
+  @computed("channels.publicChannels")
```

Comment:
Cant we rely on chat.publicChannels instead of all this fetching/filtering logic?

Reply:
No, we cannot show the current channel in the list that you are moving messages from, that would be confusing.

Reply:
I mean Im fine the reject part, but it could be a computed property on `chat.publicChannels` instead of all the fetching logic we have here.

```
@discourseComputed("chat.publichChannels")
avaiableChannels(publicChannels) {
  return publicChannels.reject(...);
}
```